packages/gitbook/src/components/PageIcon/PageIcon.tsx

```diff
@@ -0,0 +1,27 @@
+import { RevisionPage } from '@gitbook/api';
+import { Icon, IconName } from '@gitbook/icons';
+
+import { Emoji } from '@/components/primitives';
+import { ClassValue, tcls } from '@/lib/tailwind';
+
+export function PageIcon(props: { page: RevisionPage; style?: ClassValue }) {
+    const { page, style } = props;
+
+    if (page.emoji) {
+        return (
+            <Emoji
+                code={page.emoji}
+                style={[
+                    style,
+                    // We reset the color that could be passed as "style"
+                    // as emojis should always be rendered with normal text opacity
+                    'text-inherit',
```

Comment:
curious why we aren't just using className here?

Reply:
This repository is using `style` as a name for the values that are compiled as class names. This thing will become a className